crypto/src/dex/swap.rs

```diff
@@ -38,6 +38,7 @@ pub fn generate_swap_asset_id(
     delta_1: u64,
     delta_2: u64,
     fee: u64,
+    // TODO: try passing Address type here instead of the parts
```

Comment:
We should probably sleep on this until after adding the clue key to the `Note`, so that a `Note` is literally just a typed `Value` plus the `Address` it was sent to.